src/api/helpers/map-and-filter-ordnance-survey-addresses/index.ts

```diff
@@ -0,0 +1,37 @@
+import { OrdnanceSurveyResponse, Address } from '../../types';
+import mapAddress from '../map-ordnance-survey-address';
+
+/**
+ * mapAndFilterOrdnanceSurveyAddresses
+ * filters address based on building name and house name/number
+ * maps filtered address to correct format
+ * returns array of matching addresses
+ * @param {String} house name or number
+ * @param {Array<OrdnanceSurveyResponse>} ordnanceSurveyResponse
+ * @returns {Array<Address>} mapped addresses
+ */
+const mapAndFilterOrdnanceSurveyAddresses = (houseNameOrNumber: string, ordnanceSurveyResponse: Array<OrdnanceSurveyResponse>): Array<Address> => {
+  try {
+    console.info('Mapping and filtering Ordnance Survey addresses');
+
+    const mappedAndFiltered = [] as Array<Address>;
+
+    ordnanceSurveyResponse.forEach((address) => {
+      if (address.DPA.SUB_BUILDING_NAME && address.DPA.SUB_BUILDING_NAME.includes(houseNameOrNumber)) {
```

Comment:
does the first one need optional chaining?

Reply:
The first one? I'm not following